cliccl/load.go: change `load show summary` output format

Elliebababa opened the pull request:
Previously, `load show summary` output the results in an
unstructed way, which made it harder to filter information.
This patch refactors the output of `load show summary` and
output the information in json format. In this way, the output
is easier to handle and can be filtered out information through other
command-line JSON processor.

Release note (cli change): `load show summary` used to output
the results in an unstructed way, which made it harder to filter
information. We refactor the output of `load show summary` and
output the information in json format. In this way, the output
is easier to handle and can be filtered out information through other
command-line JSON processor.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/63100)
<!-- Reviewable:end -->

pbardea commented on pkg/ccl/cliccl/load.go:
```diff
@@ -237,77 +243,90 @@ func runLoadShowIncremental(cmd *cobra.Command, args []string) error {
+		BuildInfo           string
+		Files               []backupFileDisplayMsg
+		Spans               string
+		DataBaseDescriptors map[descpb.ID]string
```
nit: `DatabaseDescriptors` rather than `DataBaseDescriptors`

pbardea reviewed (approved):
Nice! Looking good

Elliebababa committed:
cliccl/load.go: change `load show summary` output format

Previously, `load show summary` output the results in an
unstructed way, which made it harder to filter information.
This patch refactors the output of `load show summary` and
output the information in json format. In this way, the output
is easier to handle and can be filtered out information through other
command-line JSON processor.

Release note (cli change): `load show summary` used to output
the results in an unstructed way, which made it harder to filter
information. We refactor the output of `load show summary` and
output the information in json format. In this way, the output
is easier to handle and can be filtered out information through other
command-line JSON processor.

Elliebababa commented:
TFTR!

bors r=pbardea

knz commented:
I think this may be a good candidate for a backport to the 21.1 branch?
